src/format_url.js

```diff
@@ -12,6 +12,7 @@ module.exports = function(path, accessToken) {
     }
 
     var url = (document.location.protocol === 'https:' || config.FORCE_HTTPS) ? config.HTTPS_URL : config.HTTP_URL;
+    url = (url.indexOf('/v4')) ? url.split('/v4')[0] : url;
```

Comment:
* We have unit tests [here](https://github.com/mapbox/mapbox.js/blob/mb-pages/test/spec/format_url.js).
* The check here should be limited to `/v4` at the start of the URL. We haven't supported `/v3` since 2.0.0; I don't think we need to start now.
* In case somebody is reading them, we should also revert `config.HTTP_URL` / `config.HTTPS_URL` to the previous values.